ssl/ssl_lib.c

```diff
@@ -560,17 +560,35 @@ void OPENSSL_VPROC_FUNC(void) {}
 
 static int clear_record_layer(SSL_CONNECTION *s)
 {
-    int ret;
+    int ret = 1;
 
-    /* We try and reset both record layers even if one fails */
+    /* Clear any buffered records we no longer need */
+    while (s->rlayer.curr_rec < s->rlayer.num_recs)
```

Comment:
should ret be included in the while condition here so that you break out of the loop if any record release fails? i.e.:
```
int ret = 1;

while (ret == 1 && s->rlayer.curr_rec < s->rlayer.num_recs)
   ret &= ssl_release_record(...)

if (ret == 0)
   return 0;

Reply:
Actually I wondered about that myself. But in the end I thought that this really should not fail - and if it does then shouldn't we make a best effort to release everything, and then only fail at the end?